test/e2e/utils/fakeapp.go

```diff
@@ -1,7 +1,7 @@
 // Unless explicitly stated otherwise all files in this repository are licensed
```

Comment:
Idealy the `test` folder should be removed and its content moved to `controllers` or `pkg`

Reply:
will move in a subsequent PR, as per the To Do section in the description of this PR.

Reply:
Got it, for me unit tests are like `controllers/replica_calculator_test.go`, which we don't need to change. These are more e2e tests as they rely on complete system running.